logviewer/src/main/java/org/togetherjava/logwatcher/views/MainLayout.java

```diff
@@ -0,0 +1,180 @@
+// Those are css names, and we don't need an extra Constants
+// Classes
+public class MainLayout extends AppLayout {
+
+    private final transient AuthenticatedUser authenticatedUser;
+    private H1 viewTitle;
+
+    public MainLayout(AuthenticatedUser authUser) {
+        this.authenticatedUser = authUser;
+        setPrimarySection(Section.DRAWER);
+        addToNavbar(true, createHeaderContent());
+        addToDrawer(createDrawerContent());
+
+        if (authUser.canRegister() && !authUser.isRegistered()) {
+            authUser.register();
+        }
+
+        final Optional<User> user = this.authenticatedUser.get();
+        if (user.map(User::getRoles).map(Set::isEmpty).orElse(true)) {
+            authUser.logout();
+        }
+
+    }
+
+    private static RouterLink createLink(MenuItemInfo menuItemInfo) {
+        RouterLink link = new RouterLink();
+        link.addClassNames("flex", "mx-s", "p-s", "relative", "text-secondary");
+        link.setRoute(menuItemInfo.view());
+
+        Span icon = new Span();
+        icon.addClassNames("me-s", "text-l");
+        if (!menuItemInfo.iconClass().isEmpty()) {
+            icon.addClassNames(menuItemInfo.iconClass());
+        }
+
+        Span text = new Span(menuItemInfo.text());
+        text.addClassNames("font-medium", "text-s");
+
+        link.add(icon, text);
+        return link;
+    }
+
+    private Component createHeaderContent() {
+        DrawerToggle toggle = new DrawerToggle();
+        toggle.addClassName("text-secondary");
+        toggle.addThemeVariants(ButtonVariant.LUMO_CONTRAST);
+        toggle.getElement().setAttribute("aria-label", "Menu toggle");
+
+        viewTitle = new H1();
+        viewTitle.addClassNames("m-0", "text-l");
+
+        Header header = new Header(toggle, viewTitle);
+        header.addClassNames("bg-base", "border-b", "border-contrast-10", "box-border", "flex",
+                "h-xl", "items-center", "w-full");
+        return header;
+    }
+
+    private Component createDrawerContent() {
+        H2 appName = new H2("Logviewer");
+        appName.addClassNames("flex", "items-center", "h-xl", "m-0", "px-m", "text-m");
+
+        com.vaadin.flow.component.html.Section section = new com.vaadin.flow.component.html.Section(
+                appName, createNavigation(), createFooter());
+        section.addClassNames("flex", "flex-col", "items-stretch", "max-h-full", "min-h-full");
+        return section;
+    }
+
+    private Nav createNavigation() {
+        Nav nav = new Nav();
+        nav.addClassNames("border-b", "border-contrast-10", "flex-grow", "overflow-auto");
+        nav.getElement().setAttribute("aria-labelledby", "views");
+
+        H3 views = new H3("Views");
+        views.addClassNames("flex", "h-m", "items-center", "mx-m", "my-0", "text-s",
+                "text-tertiary");
+        views.setId("views");
+
+        for (RouterLink link : createLinks()) {
+            nav.add(link);
+        }
```

Comment:
If you want, you can rewrite this as:
```suggestion
        createLinks().forEach(nav::add);
```

Reply:
Changed it to this